src/Compilers/CSharp/Portable/BoundTree/BoundExpression.cs

```diff
@@ -60,6 +60,7 @@ internal bool NeedsToBeConverted()
                 case BoundKind.UnconvertedConditionalOperator:
                 case BoundKind.DefaultLiteral:
                 case BoundKind.UnconvertedInterpolatedString:
+                case BoundKind.UnconvertedCollectionLiteralExpression:
```

Comment:
Do we have any tests for collection literals with errors that contain some of these other expressions? In particular, we should have a test that pulls on something that will crash if it (and the nested expressions) are not successfully converted, such as IOperation.